packages/ui-pattern-block/src/UIPatternBlock.tsx

```diff
@@ -1,25 +1,238 @@
 
-import { type Settings, sandpackThemeValues } from './types';
+import { Alignment, Height, type Settings, sandpackThemeValues } from './types';
+import { CodeEditor } from './components/CodeEditor';
+import {
+    DEFAULT_BLOCK_SETTINGS,
+    EDITOR_CLASSES,
+    centeringCss,
+    getBackgroundCss,
+    getCssToInject,
+    getDefaultFilesOfTemplate,
+    getHeightStyle,
+    getPaddingStyle,
+    getRadiusStyle,
+    getScriptToInject,
+    initialActiveFile,
+} from './helpers';
+import { useDebounce } from './hooks';
+import { Captions, ExternalDependencies, NPMDependencies } from './components';
+import { ResponsivePreview } from './components/ResponsivePreview';
 
 export const UIPatternBlock = ({ appBridge }: BlockProps): ReactElement => {
-    const [blockSettings] = useBlockSettings<Settings>(appBridge);
-    const { sandpackTemplate, sandpackTheme } = blockSettings;
+    const [blockSettings, setBlockSettings] = useBlockSettings<Settings>(appBridge);
+    const {
+        sandpackTemplate,
+        sandpackTheme,
+        files,
+        alignment,
+        dependencies: blockDependencies,
+        paddingChoice,
+        paddingCustom,
+        hasCustomPadding,
+        heightChoice,
+        customHeightValue,
+        isCustomHeight,
+        showResetButton,
+        showSandboxLink,
+        showResponsivePreview,
+        showCode,
+        showNpmDependencies,
+        showExternalDependencies,
+        shouldCollapseCodeByDefault,
+        shouldCollapseDependenciesByDefault,
+        backgroundColor,
+        hasBackground,
+        borderColor,
+        borderStyle,
+        borderWidth,
+        hasBorder,
+        hasRadius,
+        radiusChoice,
+        radiusValue,
+        description,
+        title,
+    } = { ...DEFAULT_BLOCK_SETTINGS, ...blockSettings };
+    const { debounce } = useDebounce();
+    const isEditing = useEditorState(appBridge);
+    const [isResponsivePreviewOpen, setIsResponsivePreviewOpen] = useState(false);
+    const [dependencies, setDependencies] = useState(blockDependencies);
+    const [reset, setReset] = useState(false);
+
+    const cssToInject = useMemo(() => {
+        const alignmentCss = alignment === Alignment.Center ? centeringCss : '';
+        const backgroundCss = hasBackground ? getBackgroundCss(backgroundColor) : '';
+        const hasAutoHeight = !isCustomHeight && heightChoice === Height.Auto;
+        return getScriptToInject(getCssToInject(hasAutoHeight, alignmentCss, backgroundCss));
+    }, [hasBackground, backgroundColor, alignment, isCustomHeight, heightChoice]);
+
+    const templateFiles = useMemo(
+        () => ({ ...getDefaultFilesOfTemplate(sandpackTemplate), ...files?.[sandpackTemplate] }),
+        [sandpackTemplate, cssToInject, dependencies, reset],
+    );
+
+    const onResetToDefault = () => {
+        setReset(!reset);
+    };
+
+    const onCodeChange = (filename: string, code: string) => {
+        if (!isEditing) {
+            return;
+        }
```

Comment:
Why is this needed?

Reply:
so that the changes that viewers make are not saved to the database.